version: '3'
services:
  grafana:
    image: grafana/grafana:5.0.3
    network_mode: host
    restart: always
    ports:
      - 3000:3000
    volumes:
      - /var/data/grafana:/var/lib/grafana
    environment:
      GF_LOG_MODE: console
      GF_LOG_LEVEL: warn
      GF_SERVER_ROOT_URL: "http://grafana.{{ env_domain }}/"
      GF_AUTH_ANONYMOUS_ENABLED: "true"
      GF_AUTH_ANONYMOUS_ORG_ROLE: Editor
{% if grafana_alerts_aws_id != "" %}
      GF_EXTERNAL_IMAGE_STORAGE_PROVIDER: s3
      GF_EXTERNAL_IMAGE_STORAGE_S3_BUCKET_URL: "{{ grafana_alerts_s3_bucket }}"
      GF_EXTERNAL_IMAGE_STORAGE_S3_ACCESS_KEY: "{{ grafana_alerts_aws_id }}"
      GF_EXTERNAL_IMAGE_STORAGE_S3_SECRET_KEY: "{{ grafana_alerts_aws_key }}"
{% endif %}
      SERVICE_NAME: system/grafana
      SERVICE_CHECK_TCP: "true"
